testsuite/testsuite.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+
+"""
+Usage::
+
+    testsuite.py [OPTIONS]
+
+Run the CUDA testsuite.
+"""
+import glob
+import os
+import shutil
+import subprocess
+import sys
+from e3.testsuite import Testsuite, logger
+from e3.testsuite.driver.classic import TestSkip, TestAbortWithError, TestAbortWithFailure
+from e3.testsuite.driver.classic import ClassicTestDriver
+from e3.anod.helper import Make
+from e3.anod.loader import spec
+from e3.anod.spec import Anod
+from e3.env import Env
+from e3.fs import mv, sync_tree, rm, mkdir
+from e3.os.fs import cd
+from e3.os.process import Run, wait_for_processes
+from e3.log import log_config, add_log_handlers
+import e3.env
+import logging
+import os
+import sys
+import signal
+from os import path
+from pathlib import Path
+
+
+ROOT = Path(sys.argv[0]).resolve().parents[1]
+EXAMPLES = ROOT / "examples"
+
+
+class CUDAExamplesDriver(ClassicTestDriver):
+
+    def check_file(self, path):
+        assert Path(path).is_file, f"Missing file: {path}"
+
+    def do_run(self):
+        TEST_DIR = EXAMPLES / self.test_env["input_directory"] 
+        sync_tree(str(TEST_DIR), self.working_dir())
+        self.check_file(Path(self.working_dir()) / "Makefile")
```

Comment:
Hi Ghjuvan, sorry for the very long delay answering. The sync has been copied from the Anod spec, which is itself doubtful. I'll fix that.

Reply:
Ok got it, the reason why it's done like this is twofold:
1. We don't want our test-generated obj files to pollute the original source (which is defined as `input_source` in the test config yaml)
2. We don't want the source to depend from its current path, because it's a driver for examples and those should be standalone if possible

So what we are doing is copying the files to the current (temporary) working directory prior to start the compilation. This does not solve the issues of a potential `obj/` directory that would start not empty, or of an example that would use absolute paths.